File: api/models/event.js
const mongoose 	= require('mongoose');
const Schema 	= mongoose.Schema;
const User		= require('../models/user');
const Playlist 		= require('../models/playlist');

const Event = new Schema({
	creator: {
		type: Schema.Types.ObjectId,
		ref: 'user',
		index: true
	},
	title:{type: String, default: "Aucun"},
	description:{type: String, default: "à définir"},
	location: {
		address : {
			p: {type: String},
			v: {type: String},
			cp: {type: String},
			r: {type: String},
			n: {type: String}
		},
		coord: {
			lat: {type: Number, required: true},
			lng: {type: Number, required: true}
		}
	},
	public: {type: Boolean, default: false},
	distance_max: {type:Number, default: 2},
	is_start:  {type: Boolean, default: false},
	is_finish:  {type: Boolean, default: false},
	distance_required: {type: Boolean, default: false},
	creation_date: {type: Date, default: Date.now},
	event_date: {type: Date, default: Date.now},
	playlist: {type: Playlist.schema},
	members : [{
		type: Schema.Types.ObjectId,
		ref: 'user'
	}],
	adminMembers: [{
		type: Schema.Types.ObjectId,
		ref: 'user'
	}],
	picture: {type: String, default: "default.jpeg"}

}, { versionKey: false });

module.exports = mongoose.model('event', Event);

File: app-ios/MusicRoom/MusicRoom/Modele/Event.swift
//
//  Event.swift
//  MusicRoom
//
//  Created by Etienne TRANCHIER on 11/19/18.
//  Copyright © 2018 Etienne Tranchier. All rights reserved.
//

import Foundation

struct Address : Codable {
    let p : String
    let v : String
    let cp : String
    let r : String
    let n : String
}

struct Coord : Codable {
    let lat : Double
    let lng : Double
}

struct Location : Codable {
    let address : Address
    let coord : Coord
}

struct Event : Codable {
    let _id : String?
    let creator : User?
    let title : String
    let description : String
    let location : Location
    let visibility : Int?
    let shared : Bool
    let creationDate : String
    let date : String
    let playlist : Playlist?
    var members : [User]
    var adminMembers : [User]
    let picture : String?
    
    enum CodingKeys : String, CodingKey{
        case shared = "public"
        case creationDate = "creation_date"
        case date = "event_date"
        case creator, title, description, location, visibility, playlist, members, adminMembers, picture, _id
    }

}

struct DataEvent : Codable {
    let myEvents : [Event]
    let friendEvents : [Event]
    let allEvents : [Event]
}
